src/include/test/unit_concepts_vs.cc

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file   unit_concepts.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2023 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * Test application of concepts with TileDB-Vector-Search types
+ *
+ */
+
+#include <catch2/catch_all.hpp>
+
+#include "concepts.h"
+#include "cpos.h"
+#include "detail/linalg/matrix.h"
+#include "detail/linalg/tdb_matrix.h"
+#include "detail/linalg/tdb_partitioned_matrix.h"
+#include "detail/linalg/vector.h"
+
+TEST_CASE("concepts_vs: test test", "[concepts_vs]") {
+  REQUIRE(true);
+}
+
+// Concepts:
+// 1. subscriptable_range
+// 2. callable
+// 3. callable_range
+// 4. dimensionable
+// 5. vectorable
+// 6. partitionable
+// 7. feature_vector
+// 8. feature_vector_array
+// 9. contiguous_feature_vector_range
+
+TEST_CASE("concepts_vs: Vector", "[concepts_vs]") {
+  CHECK(subscriptable_range<Vector<int>>);
+  CHECK(subscriptable_range<Vector<double>>);
+  CHECK(subscriptable_range<Vector<bool>>);
+
+  CHECK(callable_range<Vector<int>>);
+  CHECK(callable_range<Vector<double>>);
+  CHECK(callable_range<Vector<bool>>);
+
+  CHECK(dimensionable<Vector<int>>);
+  CHECK(dimensionable<Vector<double>>);
+  CHECK(dimensionable<Vector<bool>>);
+
+  CHECK(vectorable<Vector<int>>);
+  CHECK(vectorable<Vector<double>>);
+  CHECK(vectorable<Vector<bool>>);
+
+  CHECK(feature_vector<Vector<int>>);
+  CHECK(feature_vector<Vector<double>>);
+  CHECK(feature_vector<Vector<bool>>);  // @todo ? Vector of bool is not
+                                        // contiguous?
```

Comment:
(Minor) Could you add some more context here about what the todo is here (and everywhere below)? As it reads, I'm not sure if it's a todo or not, which is a bit confusing.

Reply:
Added a comment:  "There is an open question about whether we should consider Vector of bool bo be a contiguous range or not.   It is not contiguous in the sense that it is not a contiguous array of bools, but it is contiguous in the sense that  it is a contiguous array of bits.  For now, if it passes the concept check, we'll let it be."